.github/workflows/test_pr_changes.yml

```diff
@@ -12,6 +12,129 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  detect-changes-new:
```

Comment:
This workflow will run in parallel with the existing one in order to compare them. It's a copy of `run-tests` workflow from `_reusable_run_tests.yaml`.